internal/fs/fs.go

```diff
@@ -2148,6 +2154,13 @@ func (fs *fileSystem) OpenDir(
 	fs.handles[handleID] = handle.NewDirHandle(in, fs.implicitDirs)
 	op.Handle = handleID
 
+	// Invalidates the kernel dir-cache once the last cached response is out of
+	// kernelDirCacheTTL.
+	op.KeepCache = !in.ShouldInvalidateKernelDirCache(fs.kernelDirCacheTTL)
```

Comment:
I think we can avoid this check when cache is disabled ?

Reply:
+1